deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -369,13 +392,70 @@ public static void waitForPodActivity(ApiClient apiClient, V1PodHandler podHandl
             throw new HyscaleException(DeployerErrorCodes.FAILED_TO_INITIALIZE_POD);
         }
     }
-
-    private void deltaWait(long milliSeconds) {
-        try {
-            Thread.sleep(milliSeconds);
-        } catch (InterruptedException e) {
+    
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
```

Comment:
Should we have this method in V1PodHandler, since it is watching pods only? Like we have the tailLogs method.

Reply:
His point is right ...watch can go to V1PodHandler.
It's because we are watching on the pods .